docs/move-out-of-stock-products-to-the-end-of-a-collection/script.liquid

```diff
@@ -6,82 +6,107 @@
-      {% assign collections[collections.size] = some_collection %}
+  {% assign shop = shop_json | parse_json %}
+{% endif %}
+
+{% for collection in shop.collections %}
+  {% assign collection_id_string = "" | append: collection.id %}
+
+  {% if collection_titles_or_ids_to_include != blank  %}
+    {% unless collection_titles_or_ids_to_include contains collection_id_string
+      or collection_titles_or_ids_to_include contains collection.title %}
+      {% continue %}
+    {% endunless %}
+
+  {% elsif collection_titles_or_ids_to_exclude != blank  %}
+    {% if collection_titles_or_ids_to_exclude contains collection_id_string
+      or collection_titles_or_ids_to_exclude contains collection.title %}
+      {% continue %}
     {% endif %}
-  {% endfor %}
+  {% endif %}
 
-  {% for collection in collections %}
-    {% action "shopify" %}
-      mutation {
-        collectionUpdate(
-          input: {
-            id: {{ collection.admin_graphql_api_id | json }}
-            sortOrder: {{ options.base_sort_order__required }}
-          }
-        ) {
-          userErrors {
-            field
-            message
+  {% if collection.sort_order != "manual" %}
+    {% if force_manual_sorting_on_collections or event.preview %}
+      {% action "shopify" %}
+        mutation {
+          collectionUpdate(
+            input: {
+              id: {{ collection.admin_graphql_api_id | json }}
+              sortOrder: MANUAL
+            }
+          ) {
+            userErrors {
+              field
+              message
+            }
           }
         }
-      }
-    {% endaction %}
+      {% endaction %}
 
-    {% action "event" %}
-      {
-        "topic": "user/reorder_collection/stage_2",
-        "task_id": {{ task.id | json }},
-        "data": {
-          "collection_id": {{ collection.admin_graphql_api_id | json }}
-        }
-      }
-    {% endaction %}
-  {% endfor %}
-{% elsif event.topic == "user/reorder_collection/stage_2" %}
-  {% if event.preview %}
-    {% capture event_json %}
-      {
-        "preview": true,
-        "data": {
-          "collection_id": "gid://shopify/Collection/1234567890"
-        }
-      }
-    {% endcapture %}
+    {% else %}
+      {% log %}
+        {{ collection.title | json | append: " is not configured for manual sorting; skipping." | json }}
+      {% endlog %}
 
-    {% assign event = event_json | parse_json %}
+      {% continue %}
+    {% endif %}
   {% endif %}
 
   {% assign in_stock_product_ids = array %}
   {% assign out_of_stock_product_ids = array %}
 
   {% assign cursor = nil %}
+
   {% for n in (0..100) %}
     {% capture query %}
       query {
-        collection(id: {{ event.data.collection_id | json }}) {
+        collection(id: {{ collection.admin_graphql_api_id | json }}) {
           products(
-            sortKey: COLLECTION_DEFAULT
+            sortKey: {{ product_sort_order }}
```

Comment:
Wow. One line to dramatically simplify the runtime of this thing. I'm so happy. 😭